pkg/autodiscovery/listeners/kube_endpoints.go

```diff
@@ -175,16 +214,26 @@ func (l *KubeEndpointsListener) isEndpointsAnnotated(kep *v1.Endpoints) bool {
 	return false
 }
 
-func (l *KubeEndpointsListener) createService(kep *v1.Endpoints, firstRun bool) {
+func isServiceEndpointsAnnotated(ksvc *v1.Service) bool {
+	if ksvc != nil {
+		if _, found := ksvc.Annotations[kubeEndpointsAnnotationFormat]; found {
+			return true
+		}
+	}
+	return false
+}
+
+func (l *KubeEndpointsListener) createService(kep *v1.Endpoints, alreadyExistingService, checkServiceAnnotations bool) {
 	if kep == nil {
 		return
 	}
-	if !l.isEndpointsAnnotated(kep) {
+
+	if checkServiceAnnotations && !l.isEndpointsAnnotated(kep) {
 		// Ignore endpoints with no AD annotation on their corresponding service
```

Comment:
could you update the corresponding comment here ?

Reply:
Done